onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,335 @@
+    std::unique_ptr<IndexedSubGraph> sub_graph = std::make_unique<IndexedSubGraph>();
+
+    // bool newSubGraph(true);
+    bool isGraphSupported = true;
+    for (auto index : node_indexes) {
+
+        auto node = graph_viewer.GetNode(index);
+
+        auto layer = openvino_ep::OpenVINOLayer(node->OpType());
+
+        if (layer.getName() == "NotSupported" && layer.getOpsetVersion() < opset_version) {
+
+            isGraphSupported = false;
+            return result;
+        }
+        //Gemm, BatchNorm, Conv and Reshape cant take more than 1 input
+        if(node->OpType() == "Gemm" || node->OpType() == "BatchNormalization" || node->OpType() == "Conv" || node->OpType() == "Reshape" || node->OpType() == "MatMul"){
+
+            int count = 0;
+            for(auto input : node->InputDefs()){
+                auto name = input->Name();
+                auto it = initializers.find(name);
+                if(it == initializers.end()){
+                    count++;
+                }
+            }
+            if(count > 1){
+                return result;
+            }
+        }
+        //Dropout or Identity can't have graph inputs
+        if(node->OpType() == "Dropout" || node->OpType() == "Identity" || node->OpType() == "Concat") {
+
+            auto graph_inputs = graph_viewer.GetInputs();
+            for(auto input : node->InputDefs()){
+                auto it = find(graph_inputs.begin(), graph_inputs.end(), input);
+                if(it != graph_inputs.end()){
+                    return result;
+                }
+            }
+        }
+
+        if(node->OpType() == "MaxPool" || node->OpType() == "AveragePool"){
+            auto attributes = node->GetAttributes();
+            auto auto_pad = attributes["auto_pad"].s();
+            if(auto_pad == ""){
+                return result;
+            }
+            if(node->OutputDefs().size() > 1){
+                return result;
+            }
+        }
+        //Transpose with no attr is not supported
+        if(node->OpType() == "Transpose"){
+            auto attributes = node->GetAttributes();
+            auto perm = attributes["perm"].ints();
+            if(perm.size() == 0){
+                return result;
+            }
+
+            const auto* type_proto = node->InputDefs()[0]->TypeAsProto();
+            if(type_proto->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_STRING){
+                return result;
+            }
+        }
+
+        if(node->OpType() == "Softmax"){
+            auto attributes = node->GetAttributes();
+            auto axis = attributes["axis"].i();
+            if(axis < 0)
+                return result;
+        }
+
+    }
+    std::set<const onnxruntime::NodeArg*> fused_inputs, fused_outputs;
+
+
+    if(isGraphSupported){
+
+        auto model_proto = GetModelProtoFromFusedNode(graph_viewer);
+        SaveModel(model_proto,"ov_model.onnx");
```

Comment:
why do we need to save the model proto to a single file name?
and this means we can't run multiple models simultaneously?

Reply:
Hey George
This was for debug purposes. We changed this and now we are not saving the model anymore.